kin-sdk/KinClient.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Text.RegularExpressions;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Net.Http;
+using kin_base;
+
+namespace kin_sdk
+{
+    public class KinClient
+    {
+        private const string DefualtAppId = "anon";
+        private readonly TimeSpan TransactionTimeout = TimeSpan.FromSeconds(30);
+        private Server server;
+        internal readonly TransactionSender transactionSender;
+        internal readonly AccountInfoRetriver accountInfoRetriver;
+        internal readonly GeneralBlockchainInfoRetriever generalBlockchainInfoRetriever;
+        public readonly Environment environment;
+        public readonly string appId;
+        public readonly IKeyStoreProvider keyStore;
+
+        /// <summary>
+        /// Build a KinClient object
+        /// </summary>
+        /// <param name="environment">The environment to connect to</param>
+        /// <param name="keyStore">An implementation of the IKeyStoreProvider interface</param>
+        /// <param name="appId">A 3/4 charcter string (letters/digits) which represent the application id to add to each transaction</param>
+        /// <param name="httpClient">Optional httpClient to use to communicate with the blockchain</param>
+        public KinClient(Environment environment, IKeyStoreProvider keyStore, string appId = DefualtAppId, HttpClient httpClient=null)
```

Comment:
Typically only one HttpClient is required throughout runtime as the objects memory footprint is rather large.  I'd opt for using a HttpMessageHandler. 

HttpMessageHandler is middleware that can be used to handle request/response.  Here's an example of a HttpMessageHandler implementation.

```
 public class HttpLoggerHandler : DelegatingHandler
    {
        private readonly ILogger _logger;

        public HttpLoggerHandler(HttpMessageHandler innerHandler, ILoggerFactory loggerFactory) : base(innerHandler)
        {
           _logger = loggerFactory.CreateLogger<HttpLoggerHandler>();
        }

        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
        {
            try
            {
                _logger.LogDebug($"{request.Method} Request sent to {request.RequestUri.PathAndQuery}");

                var response = await base.SendAsync(request, cancellationToken).ConfigureAwait(false);

                _logger.LogDebug(
                    $"{request.Method} ({response.StatusCode}) Response from {request.RequestUri.PathAndQuery}");

                return response;
            }
            catch (WebException wex)
            {
                wex.Response?.Dispose();
                _logger.LogError(wex, wex.Message);
                throw;
            }
        }
    }
```

Reply:
The base library needs a httpClient object, and id prefer not to change it for now.